### PR TITLE
Styling and component restructuring

### PR DESCRIPTION
**Pull request by MargaretKrutikova**

### CODE REVIEW

File: src/components/Grid.re [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  | Container
+  | Item;
+
+let make = (~type_: gridType, children) => {
+  let getGridStyles = (): list(rule) => {
+    switch (type_) {
+    | Container =>
+      [display(`flex)]
+      @ gridContainer(Theme.spacingPx.small)
+      @ [media(Breakpoints.up(Md), gridContainer(Theme.spacingPx.medium))]
```
Comment (Yakimych) [inline src/components/Grid.re:22]: `gridContainer()` returns an array, right? So is it an array inside of an array?

File: src/components/ShoppingItem.re [hunk 2]
```diff
@@ -0,0 +1,36 @@
+open Types;
+
+let component = ReasonReact.statelessComponent("ShoppingItem");
+
+let make = (~item: shoppingItem, ~onItemChange, ~id: int, _children) => {
+  let handleProductChange = event => {
```
Comment (Yakimych) [inline src/components/ShoppingItem.re:6]: I think you should be able to annotate the function as follows:
`let handleProductChange = event: unit => {`
and avoid `();` at the end.

File: src/components/ShoppingItem.re [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    let newItem = {...item, product: ReactEvent.Form.target(event)##value};
+    onItemChange(id, newItem);
```
Comment (Yakimych) [inline src/components/ShoppingItem.re:8]: Even better if `ignore` works:
`onItemChange(id, newItem) |> ignore`

File: src/components/ShoppingList.re [hunk 4]
```diff
@@ -1,138 +1,23 @@
...
-                 }
-               />
-             </div>
-           )
+let component = ReasonReact.statelessComponent("ShoppingList");
+
+let make = (~list: shoppingList, ~onItemChange, _children) => {
+  ...component,
+  render: _self =>
+    <>
```
Comment (Yakimych) [inline src/components/ShoppingList.re]: Do we need the Fragment here? Looks like everything is already wrapped in a div.

File: src/components/TextField.re [hunk 5]
```diff
@@ -0,0 +1,48 @@
+let component = ReasonReact.statelessComponent("Input");
```
Comment (Yakimych) [inline src/components/TextField.re]: Should this be "TextField" rather than "Input" to match the file name?

File: src/components/TextField.re [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          ),
+        ])
+      )>
+      <input
+        type_="text"
+        placeholder
+        value
+        onChange
+        className=Css.(
+          style([
```
Comment (Yakimych) [inline src/components/TextField.re:23]: Would be nice to throw in `maxLength` here too - probably as a prop with default value?

File: src/pages/ShoppingListPage.re [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+        let items = Array.append(state.shoppingList.items, [|newItem|]);
+        ReasonReact.Update(updateItems(state, items));
+
+      | ChangeShoppingItem(index, newItem) =>
+        let updateFn = _item => newItem;
+
+        let items =
+          updateWithIndex(state.shoppingList.items, updateFn, index);
+        ReasonReact.Update(updateItems(state, items));
```
Comment (Yakimych) [inline src/pages/ShoppingListPage.re:87]: Nice! I can show you how to make the reducer testable later if you'd like.

File: src/theme/Breakpoints.re [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  switch (bp) {
+  | Xs => 0
+  | Sm => 600
+  | Md => 960
+  | Lg => 1280
+  | Xl => 1920
+  };
+
+let up = (bp: breakpoint) =>
+  "(min-width: " ++ string_of_int(breakpointToPx(bp)) ++ "px)";
```
Comment (Yakimych) [inline src/theme/Breakpoints.re:18]: This looks cool - so it supports all kind of screen sizes?

File: src/theme/Theme.re [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  xlarge: int,
+};
+
+type themeColors = {
+  greyDefault: color,
+  greySecondary: color,
+  bgColor: color,
+};
+
+let spacingUnitPx = 6;
```
Comment (Yakimych) [inline src/theme/Theme.re:18]: Yep - I already like this CSS-in-JS stuff 😃

**Review by Yakimych — CHANGES_REQUESTED** [review]

File: src/theme/Breakpoints.re (hunk 8 above)
Comment (MargaretKrutikova) [reply src/theme/Breakpoints.re:18]: Yes! :)

File: src/components/Grid.re (hunk 1 above)
Comment (MargaretKrutikova) [reply src/components/Grid.re:22]: Yeah, when you want to use `media` you need to send in an array of styles that will be applied to that breakpoint, so eventually you will have an outer array with styles not applied to any breakpoints and breakpoints where each of them has its internal array of styles.

File: src/components/ShoppingItem.re (hunk 2 above)
Comment (MargaretKrutikova) [reply src/components/ShoppingItem.re:6]: Annotated it in `ShoppingList`.

File: src/components/TextField.re (hunk 6 above)
Comment (MargaretKrutikova) [reply src/components/TextField.re:23]: Done :)

File: src/components/ShoppingList.re (hunk 4 above)
Comment (MargaretKrutikova) [reply src/components/ShoppingList.re]: Removed it :)

**Review by Yakimych — APPROVED** [review]

**Pull request merged**